### ISSUE TITLE
2bit file processing is slow if it contains many refseqs

### ISSUE THREAD

**Issue by cmdcolin** [GMOD/jbrowse#1116]
The 2bit file parsing currently launches a Promise-per-refseq on initial parse which becomes very slow when there are many refseqs

**Comment by rbuels** [comment]
my 2c, i'd say this is more of a bug than a tech debt things, cause it's a scalability issue

**Comment by cmdcolin** [comment]
Agree :)

**Comment by rbuels** [comment]
implementation sketch:
* break out as a node module and refactor

**Comment by rbuels** [comment]
@cmdcolin do you have a problematic 2bit file I can test with?

**Comment by cmdcolin** [comment]
I forgot to mention that this happens when the 2bit file is used as a reference instead of refSeqs.json

If you use prepare-refseqs.pl currently it writes out a refSeqs.json generated from the twobit file which makes twobit loading fast, but if you use the "Open sequence file" or create a trackList.json like this

```
{
   "formatVersion" : 1,
   "tracks" : [
      {
         "category" : "Reference sequence",
         "key" : "Reference sequence",
         "label" : "DNA",
         "seqType" : "dna",
         "storeClass" : "JBrowse/Store/Sequence/TwoBit",
         "type" : "SequenceTrack",
         "urlTemplate" : "seq/out.2bit",
         "useAsRefSeqStore" : 1
      }
   ],
   "refSeqs": "seq/out.2bit"
}
```

Here's a simple tar.gz of a data dir with this

[data.tar.gz](https://github.com/GMOD/jbrowse/files/2223680/data.tar.gz)

It has 10000 refseqs but takes 30 seconds to load on my computer due to tons and tons of readSequenceHeader calls

**Comment by cmdcolin** [comment]
Fixed via #1146! Nice work
